### PR TITLE
rename confusing getMailHeaderColor to getColorPrimary, ref #3491

### PR DESCRIPTION
**Pull request by jancborchardt**
Fix confusion caused by the oddly specific variable name, ref #3491

Please review @leonklingele @fancycode @nextcloud/theming :)

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Yay 💥 all the themes 🙈

**Comment by jancborchardt** [comment]
Goddamnit … @schiessle anything we can do to migrate this? That variable name is really stupid.

**Comment by MorrisJobke** [comment]
> Goddamnit … @schiessle anything we can do to migrate this? That variable name is really stupid.

Just put it in the release notes. I'm fine with that and it's not that hard to do.

File: lib/private/legacy/defaults.php [hunk 1]
```diff
@@ -275,9 +275,9 @@ public function buildDocLinkToKey($key) {
 	 * Returns mail header color
 	 * @return string
 	 */
-	public function getMailHeaderColor() {
-		if ($this->themeExist('getMailHeaderColor')) {
-			return $this->theme->getMailHeaderColor();
+	public function getColorPrimary() {
+		if ($this->themeExist('getColorPrimary')) {
+			return $this->theme->getColorPrimary();
```
Comment (juliushaertl) [inline lib/private/legacy/defaults.php]: Couldn't we add a fallback to getMailHeaderColor here to avoid breaking Thames?

**Comment by juliushaertl** [comment]
@MorrisJobke @jancborchardt I've added a commit that we fallback to getMailHeaderColor so existing themes will keep working.

**Comment by jancborchardt** [comment]
Also added a commit to rename defaultMailHeaderColor as well.

Good to go from my view, reviews please. :)

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by MorrisJobke** [comment]
I fixed the merge conflict BTW 😉

**Review by rullzer — APPROVED** [review]
Fine by me!

**Pull request merged**